### ISSUE TITLE
[v1] bag constructor allows for whitespace between angle brackets

### ISSUE THREAD

**Issue by alancai98** [partiql/partiql-lang-kotlin#1498]
Something I noticed when trying to add some special operator parsing rules in the v1 branch is that it currently allows whitespace between the angle brackets:

```
Welcome to the PartiQL shell!
Typing mode: LEGACY
Using version: 1.0.0-SNAPSHOT-4dd09721
PartiQL> <<1>>;
===' 
<<
  1
>>
--- 
OK!
PartiQL> < < 1 > >;
===' 
<<
  1
>>
--- 
OK!
```

whereas the previous parsing behavior on `main` and all prior releases require the angle brackets to be together:
```
Welcome to the PartiQL shell!
Typing mode: LEGACY
Using version: 0.14.5-ab65143c
PartiQL> <<1>>;
==='
<<
  1
>>
---
OK!
PartiQL> < < 1 > >;

mismatched input '<' expecting {'ANY', 'AVG', 'BIT_LENGTH', 'CASE', 'CAST', 'CHARACTER_LENGTH', 'CHAR_LENGTH', 'COALESCE', 'COUNT', 'CREATE', 'CURRENT_DATE', 'CURRENT_USER', 'DATE', 'DELETE', 'DROP', 'EVERY', 'EXCLUDED', 'EXEC', 'EXISTS', 'EXPLAIN', 'EXTRACT', 'DATE_ADD', 'DATE_DIFF', 'FALSE', 'FROM', 'INSERT', 'LOWER', 'MAX', 'MIN', 'NOT', 'NULL', 'NULLIF', 'OCTET_LENGTH', 'OVERLAY', 'POSITION', 'REPLACE', 'SELECT', 'SET', 'SIZE', 'SOME', 'SUBSTRING', 'SUM', 'TIME', 'TIMESTAMP', 'TRIM', 'TRUE', 'UPDATE', 'UPPER', 'UPSERT', 'VALUES', 'LAG', 'LEAD', 'CAN_CAST', 'CAN_LOSSLESS_CAST', 'MISSING', 'PIVOT', 'REMOVE', 'LIST', 'SEXP', '+', '-', '@', '<<', '[', '{', '(', '?', LITERAL_STRING, LITERAL_INTEGER, LITERAL_DECIMAL, IDENTIFIER, IDENTIFIER_QUOTED, ION_CLOSURE}
```

Seems like https://github.com/partiql/partiql-lang-kotlin/pull/1449 inadvertently introduced this parsing change. Will need to confirm w/ team whether this behavior should exist in the parsing.

**Comment by alancai98** [comment]
Above change in v1 also allows users to insert a comment in between the bag constructor angle brackets:

```
PartiQL> </* some comment */< 1 > >;
==='
<<
  1
>>
---
OK!
```

**Comment by alancai98** [comment]
If we deem this is unintended behavior, I have a fix as part of https://github.com/partiql/partiql-lang-kotlin/issues/1478, which looks at the hidden token stream for any hidden whitespace/comments.

**Comment by alancai98** [comment]
As brought up in Yingtao's [comment](https://github.com/partiql/partiql-lang-kotlin/pull/1500#discussion_r1664570147), there are a few other operators we'll need to correct this behavior for:

```
PartiQL> 1 < > 2;
===' 
true
--- 
OK!
PartiQL> 1 </* some comment */> 2;
===' 
true
--- 
OK!
PartiQL> 1 ! = 2;
===' 
true
--- 
OK!
PartiQL> 1 !/* some comment */= 2;
===' 
true
--- 
OK!
```

I'll include this fix in #1499.
